pkg/database/database.go

```diff
@@ -0,0 +1,84 @@
+package database
+
+import (
+	"database/sql"
+	"fmt"
+	"regexp"
+	"github.com/gobuffalo/packr/v2"
+	"github.com/golang-migrate/migrate/v4"
+	"github.com/golang-migrate/migrate/v4/source"
+	"github.com/jmoiron/sqlx"
+	sqlite3 "github.com/mattn/go-sqlite3"
+	"github.com/stashapp/stashdb/pkg/logger"
+	"github.com/stashapp/stashdb/pkg/utils"
+	"os"
+)
+
+var DB *sqlx.DB
+var appSchemaVersion uint = 1
+
+const sqlite3Driver = "sqlite3_regexp"
+
+func Initialize(databasePath string) {
+	runMigrations(databasePath)
+
+	// register custom driver with regexp function
+	registerRegexpFunc()
+
+	// https://github.com/mattn/go-sqlite3
+	conn, err := sqlx.Open(sqlite3Driver, "file:"+databasePath+"?_fk=true")
+	conn.SetMaxOpenConns(25)
+	conn.SetMaxIdleConns(4)
+	if err != nil {
+		logger.Fatalf("db.Open(): %q\n", err)
+	}
+	DB = conn
+}
+
+func Reset(databasePath string) {
+	_ = DB.Close()
+	_ = os.Remove(databasePath)
+	Initialize(databasePath)
+}
+
+// Migrate the database
+func runMigrations(databasePath string) {
+	migrationsBox := packr.New("Migrations Box", "./migrations")
+	packrSource := &Packr2Source{
+		Box:        migrationsBox,
+		Migrations: source.NewMigrations(),
+	}
+
+	databasePath = utils.FixWindowsPath(databasePath)
+	s, _ := WithInstance(packrSource)
+	m, err := migrate.NewWithSourceInstance(
+		"packr2",
+		s,
+		fmt.Sprintf("sqlite3://%s", "file:"+databasePath),
```

Comment:
should be `fmt.Sprintf("sqlite3://%s", databasePath),` probably copy paste left from sqlx invocation?

Reply:
Why are we dropping the `file:` prefix?

Reply:
It appears that the presence of `file:` meant that the subsequent path must be an absolute path. Since stashdb was by default passing a relative path to the database file, it was causing a panic on Linux. Removing the `file:` directive fixed the panic and has no apparent issues on Windows.